### PR TITLE
[Robot] Edit button should be disabled on RD status mappings page

### PR DESCRIPTION
**Pull request by rjanjanam**
This test is to ensure that on the RD default mappings page on the NPSP settings, Edit button stays disabled when the user has no custom status mappings

Few changes to address flakiness around affiliations and other tests are made as part of this PR

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Review by rponti-sforg — COMMENTED** [review]
@rjanjanam The first two files, **OpportunityPageObject.py** and **remove_secondary_affiliation**, are being updated as well.  Are these just files you have been meaning to update / commit or do they have something in common with the RD2 status mappings feature?  

The **zverify_edit_disabled_on_mappings_for_rd2.robot** file looks good from a functional perspective and what we want to test.  Yep, this will fail until the issue / bug is fixed with the Edit button currently being enabled.  Can I update the Documentation part of the robot file?

We probably should have an automation that has a 'custom' picklist value where the Edit button is then supposed to be enabled.

File: robot/Cumulus/tests/browser/affiliations/remove_secondary_affiliation.robot [hunk 1]
```diff
@@ -30,7 +30,7 @@ Remove Secondary Affiliation for Contact
     Select Tab                        Related
     Click Related Item Popup Link     Organization Affiliations    ${account}[Name]        Delete
     Wait For Modal                    New                          Affiliation             expected_heading=Delete Affiliation
-    Click Modal Button                Delete
+    Click Element                    //button/span[contains(text(),'Delete')]
```
Comment (schlossmj) [inline robot/Cumulus/tests/browser/affiliations/remove_secondary_affiliation.robot:33]: We should add this xpath to the locators file and use the Locator Manager to call it.

**Review by schlossmj — CHANGES_REQUESTED** [review]

File: robot/Cumulus/tests/browser/affiliations/remove_secondary_affiliation.robot (hunk 1 above)
Comment (rjanjanam) [reply robot/Cumulus/tests/browser/affiliations/remove_secondary_affiliation.robot]: Made the right change

**Comment by rjanjanam** [comment]
> @rjanjanam The first two files, **OpportunityPageObject.py** and **remove_secondary_affiliation**, are being updated as well. Are these just files you have been meaning to update / commit or do they have something in common with the RD2 status mappings feature?
> 
> The **zverify_edit_disabled_on_mappings_for_rd2.robot** file looks good from a functional perspective and what we want to test. Yep, this will fail until the issue / bug is fixed with the Edit button currently being enabled. Can I update the Documentation part of the robot file?
> 
> We probably should have an automation that has a 'custom' picklist value where the Edit button is then supposed to be enabled.

There are few changes to affiliations to address flaky test. We do have automation that has a 'custom' picklist value where the Edit button is then supposed to be enabled.

**Review by schlossmj — APPROVED** [review]

**Review by rponti-sforg — APPROVED** [review]
Approved.  

I did re-read the description of the PR and saw you made a note for updating **Affiliations**.  The other 'flakiness' that I typically see with Affiliations is this error, "**Expected 140qYiTA ZsXJNhyx allocation of Former was not found**".  You can see it at the bottom of this Test Execution Log: https://sfdo-metaci.herokuapp.com/tests/result/239492605

This may be a 'timing' issue and something to keep our eye on.

**Comment by rponti-sforg** [comment]
> > @rjanjanam The first two files, **OpportunityPageObject.py** and **remove_secondary_affiliation**, are being updated as well. Are these just files you have been meaning to update / commit or do they have something in common with the RD2 status mappings feature?
> > The **zverify_edit_disabled_on_mappings_for_rd2.robot** file looks good from a functional perspective and what we want to test. Yep, this will fail until the issue / bug is fixed with the Edit button currently being enabled. Can I update the Documentation part of the robot file?
> > We probably should have an automation that has a 'custom' picklist value where the Edit button is then supposed to be enabled.
> 
> There are few changes to affiliations to address flaky test. We do have automation that has a 'custom' picklist value where the Edit button is then supposed to be enabled.

Thank you @rjanjanam.  Are these WI's a duplicate of this PR: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008k2JCIAY/view
https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008gwJyIAI/view

This one slipped by me... You are correct, this is the WI for the RD2 Status to State Mapping custom picklist values:
https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008KgmpIAC/view

**Comment by rponti-sforg** [comment]
QE is now complete with testing this PR and subsequent WI.

The robot test, for now, is expected to to fail due to the Edit button being enabled when the Status to State Mapping page displays only the 'default' States (Active, Lapsed, Closed, Paused).

The robot failure that is observed is expected and has been confirmed that it is the correct failure displayed:

Element '<selenium.webdriver.remote.webelement.WebElement (session="2786d6369e7dc849da924cda3e93a172", element="0.9276641413271955-242")>' is enabled.

Proceeding with merge of this PR and updating WI accordingly.

**Pull request merged**